src/plugins/query_enhancements/public/search/ppl_search_interceptor.ts

```diff
@@ -73,8 +73,10 @@ export class PPLSearchInterceptor extends SearchInterceptor {
     const query: Query = this.queryService.queryString.getQuery();
     const dataset = query.dataset;
     if (!dataset || !dataset.timeFieldName) return query;
+    const [baseQuery, ...afterPipeParts] = query.query.split('|');
+    const afterPipe = afterPipeParts.length > 0 ? ` | ${afterPipeParts.join('|').trim()}` : '';
     const timeFilter = this.getTimeFilter(dataset.timeFieldName);
-    return { ...query, query: query.query + timeFilter };
+    return { ...query, query: baseQuery + timeFilter + afterPipe };
```

Comment:
would be good if this logic has tests, maybe later as this class doesn't have unit tests yet

Reply:
+1